src/Std/Data/DTreeMap/RawLemmas.lean

```diff
@@ -217,4 +218,104 @@ theorem size_insertIfNew_le [TransCmp cmp] (h : t.WF) {k : α} {v : β k} :
     (t.insertIfNew k v).size ≤ t.size + 1 :=
   Impl.size_insertIfNew!_le h
 
+@[simp]
+theorem get?_emptyc [TransCmp cmp] [LawfulEqCmp cmp] {a : α} :
+    (∅ : DTreeMap α β cmp).get? a = none :=
+  Impl.get?_empty
+
+theorem get?_of_isEmpty [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a : α} :
+    t.isEmpty = true → t.get? a = none :=
+  Impl.get?_of_isEmpty h
+
+theorem get?_insert [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a k : α} {v : β k} :
+    (t.insert k v).get? a =
+      if h : cmp k a = .eq then some (cast (congrArg β (compare_eq_iff_eq.mp h)) v) else t.get? a :=
+  Impl.get?_insert! h
+
+@[simp]
+theorem get?_insert_self [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {k : α} {v : β k} :
+    (t.insert k v).get? k = some v :=
+  Impl.get?_insert!_self h
+
+theorem contains_eq_isSome_get? [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a : α} :
+    t.contains a = (t.get? a).isSome :=
+  Impl.contains_eq_isSome_get? h
+
+theorem mem_iff_isSome_get? [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a : α} :
+    a ∈ t ↔ (t.get? a).isSome :=
+  Impl.mem_iff_isSome_get? h
+
+theorem get?_eq_none_of_contains_eq_false [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a : α} :
+    t.contains a = false → t.get? a = none :=
+  Impl.get?_eq_none_of_contains_eq_false h
+
+theorem get?_eq_none [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {a : α} :
+    ¬ a ∈ t → t.get? a = none :=
+  Impl.get?_eq_none h
+
+theorem get?_erase [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {k a : α} :
+    (t.erase k).get? a = if cmp k a = .eq then none else t.get? a :=
+  Impl.get?_erase! h
+
+theorem get?_erase_self [TransCmp cmp] [LawfulEqCmp cmp] (h : t.WF) {k : α} :
```

Comment:
This should be `@[simp]` to be consistent with the other similar lemmas in this file.